herbert_core/utilities/global_path/resolve_path.m

```diff
@@ -0,0 +1,23 @@
+function fullpath = resolve_path(filename)
+% Function to resolve full path to a file or a folder
+%
+%
+if ~ischar(filename)
+    error('RESOLVE_PATH:invalid_argument',...
+        'input should be a string, identifying file or path to a file. Got: %s',...
+        fevalc('disp(filename)'))
+end
+
+file=java.io.File(filename);
+fullpath = char(file.getCanonicalPath());
+if isunix && fullpath(end) == '~'
```

Comment:
this doesn't correctly handle `~` in Linux:
```
>>> cd /home/user/bob/workingDir
>>> g = java.io.File("~/test.file");
>>> g.getCanonicalPath()  # should be /home/user/bob/test.file
/home/user/bob/workingDir/~/test.file
```
The `~` should be replaced by user homedir

Reply:
Since `~` will always start the path you can check and switch before creating the file:
```
if starts_with_tilde
  file = java.io.File(userhome, filename(2:end))
else
  file = java.io.File(filename)
end
```

Reply:
Also does not work. 
In addition to that,  getuserpath on iDaaaS resolves to simulink and java(~) to real mounting path. Theoretically the same, but pain for unit tests